File: vigil-backend/app/schemas/review_queue.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict
from app.schemas.base import PaginatedResponse
from app.schemas.pull_request import PullRequestRead


class ReviewQueueItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    pull_request: PullRequestRead
    latest_analysis_id: Optional[uuid.UUID] = None
    latest_analysis_status: Optional[str] = None
    latest_review_id: Optional[uuid.UUID] = None
    latest_review_status: Optional[str] = None
    queued_at: Optional[datetime] = None


class ReviewQueueListResponse(PaginatedResponse):
    items: List[ReviewQueueItem]

File: vigil-frontend/src/types/reviewQueue.ts
import type { PaginatedResponse } from './api';
import type { PullRequestRead } from './pullRequest';

export interface ReviewQueueItem {
  pull_request: PullRequestRead;
  latest_analysis_id: string | null;
  latest_analysis_status: string | null;
  latest_review_id: string | null;
  latest_review_status: string | null;
  queued_at: string | null;
}

export type ReviewQueueListResponse = PaginatedResponse<ReviewQueueItem>;
